lib/Doctrine/DBAL/Migrations/Version.php

```diff
@@ -461,7 +460,7 @@ private function formatParamsForOutput(array $params, array $types)
             if (Type::hasType($type)) {
```

Comment:
I think that we can simplify some things, what do you think about this:

```php
private function formatParamsForOutput(array $params, array $types): string
{
    if (empty($params)) {
        return '';
    }

    $out = [];

    foreach ($params as $key => $value) {
        $formatted = $this->formatParameter($value, $types[$key] ?? 'string');

        $out[] = is_string($key) ? sprintf(':%s => %s', $key, $formatted) : $formatted;
    }

    return sprintf('with parameters (%s)', implode(', ', $out));
}

private function formatParameter($value, string $type) : string
{
    if (Type::hasType($type)) {
        return Type::getType($type)->convertToDatabaseValue(
            $value,
            $this->connection->getDatabasePlatform()
        );
    }

    // This is quite important since `$value` and `$type` are things provided
    // by the user, so we can't really ensure that `$value` will always be an array.
    if (is_array($value)) { 
        return implode(', ', array_map([$this, 'parameterToString'], $value));
    }

    return $this->parameterToString($value);
}

private function parameterToString($value) : string
{
    if (is_int($value) || is_string($value)) {
        return (string) $value;
    }

    if (is_bool($value)) {
        return $value === true ? 'true' : 'false';
    }

    return '?';
}
```

Reply:
I will take it

Reply:
We could also make `parameterToString()` recursive, if needed... like:

```php
private function formatParameter($value, string $type) : string
{
    if (Type::hasType($type)) {
        return Type::getType($type)->convertToDatabaseValue(
            $value,
            $this->connection->getDatabasePlatform()
        );
    }

    return $this->parameterToString($value);
}

private function parameterToString($value) : string
{
    if (is_array($value)) { 
        return implode(', ', array_map([$this, 'parameterToString'], $value));
    }

    if (is_int($value) || is_string($value)) {
        return (string) $value;
    }

    if (is_bool($value)) {
        return $value === true ? 'true' : 'false';
    }

    return '?';
}
```